internal/files/manifest.go

```
@@ -0,0 +1,183 @@
+	file, err := os.Open(filepath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	hash := md5.New()
+	if _, err := io.Copy(hash, file); err != nil {
+		return "", err
+	}
+
+	return hex.EncodeToString(hash.Sum(nil)), nil
+}
+
+// CompareManifests compares two manifests and returns what changed
+func CompareManifests(current, previous *FileManifest) (added, modified, deleted []string) {
+	// Create maps for efficient comparison
+	currentFiles := make(map[string]FileEntry)
+	for _, file := range current.Files {
+		currentFiles[file.Path] = file
+	}
+
+	previousFiles := make(map[string]FileEntry)
+	for _, file := range previous.Files {
+		previousFiles[file.Path] = file
+	}
+
+	// Find added and modified files
+	for path, currentFile := range currentFiles {
+		previousFile, exists := previousFiles[path]
+		if !exists {
+			added = append(added, path)
+		} else if currentFile.Hash != previousFile.Hash {
+			modified = append(modified, path)
+		}
+	}
+
+	// Find deleted files
+	for path := range previousFiles {
+		if _, exists := currentFiles[path]; !exists {
+			deleted = append(deleted, path)
+		}
+	}
+
+	return added, modified, deleted
+}
+
+// IgnoreMatcher handles file ignore patterns
+type IgnoreMatcher struct {
+	patterns []string
+}
+
+// NewIgnoreMatcher creates a new ignore matcher
+func NewIgnoreMatcher(patterns []string) *IgnoreMatcher {
+	return &IgnoreMatcher{
+		patterns: patterns,
+	}
+}
+
+// ShouldIgnore checks if a path should be ignored
+func (m *IgnoreMatcher) ShouldIgnore(path string) bool {
+	// Always ignore .git directory
+	if strings.HasPrefix(path, ".git/") || path == ".git" {
+		return true
+	}
+
+	// Always ignore .cerebrium directory
+	if strings.HasPrefix(path, ".cerebrium/") || path == ".cerebrium" {
+		return true
+	}
+
+	// Check against patterns
+	for _, pattern := range m.patterns {
+		// Simple pattern matching (can be enhanced with glob patterns)
+		if matched, _ := filepath.Match(pattern, path); matched {
```

Please don't ignore the error from Match on this and line 176. The error should be bubbled all the way up and returned to the user - last thing we want is to add or delete a file on the remote because of some silly matcher error, will be very confusing to users